opencog/guile/SchemeSmob.cc

```diff
@@ -36,7 +36,7 @@ using namespace opencog;
  * simplify that code, and probably improve performance just a bit.
  */
 
-scm_t_bits SchemeSmob::cog_misc_tag;
+std::atomic<scm_t_bits> SchemeSmob::cog_misc_tag;
```

Comment:
Hi Linas,
 I am not going to make any further changes, I think I made certain changes
which might not be sensible, and I am not sure of the right way to fix the
problem. I hope you can get head installed soon and see the problem
yourself as I have been blocked by the issue to complete my tasks for
monday demo.

On Sat, Jul 30, 2016 at 1:40 AM, Linas Vepštas <notifications@github.com>
wrote:

> In opencog/guile/SchemeSmob.cc
> <https://github.com/opencog/atomspace/pull/843#discussion_r72831451>:
>
> > @@ -36,7 +36,7 @@ using namespace opencog;
> >   * simplify that code, and probably improve performance just a bit.
> >   */
> >
> > -scm_t_bits SchemeSmob::cog_misc_tag;
> > +std::atomic<scm_t_bits> SchemeSmob::cog_misc_tag;
>
> This is a nonsense fix. cog_misc_tag is set once and only once ever,
> making it atomic cannot possibly solve any problem that you are having.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/opencog/atomspace/pull/843/files/101c6c5afe3b2f889a9101a1a3a0216b684195de#r72831451>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHM_f1Rx3ij6VpAD7hc5NEDfLU8jfK7Cks5qajsSgaJpZM4JYEV9>
> .
>